composer/utils/object_store.py

```diff
@@ -286,7 +288,27 @@ def upload_object_via_stream(self,
                                                 headers=headers)
 
     def _get_object(self, object_name: str):
-        return self._provider.get_object(self._container.name, object_name)
+        """Get object from object store. Recursively follow any symlinks.
+
+        Args:
+            object_name (str): The name of the object.
+        """
+        obj = self._provider.get_object(self._container.name, object_name)
```

Comment:
if `object_name` does not exist and does not end with `'.symlink'`, thoughts about attempting to get `object_name + ".symlink"`? This would allow the caller to omit the `'.symlink'` extension when getting objects, which would clean up the user API a bit when retrieving checkpoints from object stores (e.g. the user could specify `latest` instead of `latest.symlink`)